plutus-metatheory/src/Builtin.lagda.md

```diff
@@ -79,7 +95,127 @@ data Builtin : Set where
   mkPairData                      : Builtin
   mkNilData                       : Builtin
   mkNilPairData                   : Builtin
+```
+
+## Signatures
+
+The following module defines a `signature` function assigning to each builtin an abstract type.
+
+```
+private module SugaredSignature where
+```
+Syntactic sugar for writing the signature of built-ins.
+This is defined in its own module so that these definitions are not exported.
+
+```
+    -- number of different type variables
+    ∙ = 0
+    ∀a = 1
+    ∀b,a = 2
+
+    -- shortened names for type constants and type constructors
+    integer bool bytestring string unit Data : ∀{n} → n ⊢♯
```

Comment:
There's of course the problem that we don't have the `#` kind in the implementation. I'd like to have it, but given the amount of stuff we're currently doing it probably won't happen any time soon... I guess the hardest part would be injecting those `#` kinds into the program after GHC type checks the original Haskell code that doesn't have any `#` in it? If you've already spent some time thinking how we'd elaborate Plutus Tx with no `#` to Plutus IR with `#` in it, I'd be curious to hear.